Analysis/Tasks/trackchecks.cxx

```diff
@@ -0,0 +1,225 @@
+    hDCAxySeco_truepid->Add(3, TH2F("hDCAxySeco_truepid_pr", "DCAxy secondaries (true Pr); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxySeco_truepid->Add(4, TH2F("hDCAxySeco_truepid_de", "DCAxy secondaries (true De); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+
+    fCutDCAxy = new TF1("fMaxDCAxy", "[0]+[1]/(x^[2])", 0, 1e10); //from TPC analysis task (Run2)
+    fCutDCAxy->SetParameter(0, 0.0105);
+    fCutDCAxy->SetParameter(1, 0.0350);
+    fCutDCAxy->SetParameter(2, 1.1);
+  } //init
+
+  void process(soa::Join<aod::Collisions, aod::EvSels>::iterator const& col, soa::Join<aod::Tracks, aod::TracksExtra, aod::McTrackLabels>& tracks, aod::McParticles& mcParticles)
+  {
+
+    if (isMC) {
+      LOGF(info, "isMC=%d - Running on MC", isMC.value);
+    } else {
+      LOGF(info, "isMC=%d - Running on DATA", isMC.value);
+    }
+
+    //event selection
+    if (!isMC && !col.alias()[kINT7]) // trigger (should be skipped in MC)
+      return;
+    if (!col.sel7()) //additional cuts
+      return;
+    if (abs(col.posZ()) > 10.) // |v_z|<10cm
+      return;
+
+    //Loop on tracks
+    for (auto& track : tracks) {
+      double y = -999.;
+      int pdgcode = -999;
+      int ptype = -999;
+      bool isPrimary = false;
+
+      if (isMC) { //determine particle species base on MC truth and if it is primary or not
+        pdgcode = track.label().pdgCode();
+        if (pdgcode == 11)
+          ptype = kEl;
+        else if (pdgcode == 211)
+          ptype = kPi;
+        else if (pdgcode == 321)
+          ptype = kKa;
+        else if (pdgcode == 2212)
+          ptype = kPr;
+        else if (pdgcode == 1000010020)
+          ptype = kDe;
+
+        if (MC::isPhysicalPrimary(mcParticles, track.label())) { //is primary?
+          isPrimary = true;
+        }
+      }
+
+      if (isMC) {
+        //calculate rapidity
+        y = eta2y(track.pt(), mass[ptype], track.eta());
+
+        if (isPrimary && abs(y) < 0.5) {
+          //histograms with generated distribution (after event selection)
+          hTrkPrimAftEvSel_truepid->Fill(ptype, track.pt());
+          hTrkPrimAftEvSel->Fill(track.pt()); //charged particles
+        }
+      }
+
+      //track selection - DCAxy and DCAz cut done later - other cuts??
+      TRACKSELECTION;
+
+      //DCAxy distributions for reco, primary, secondaries (= not primary)
+      //calculate dcaxy and dcaz
+      std::pair<float, float> dcas = getdcaxyz(track, col);
+      if (isMC) {
+        if (abs(y) < 0.5) {
+          hDCAxyReco_truepid->Fill(ptype, track.pt(), dcas.first);
+          if (isPrimary)
+            hDCAxyPrim_truepid->Fill(ptype, track.pt(), dcas.first);
+          else
+            hDCAxySeco_truepid->Fill(ptype, track.pt(), dcas.first);
+        }
+      }
+
+      //Additional CUTS
```

Comment:
Why you don't use the TrackSelection class, or even better the track selection workflow and then you can use a Filter to only get accepted tracks in your task

Reply:
I wanted to use the TrackSelection class but I see that aod::TracksExtended is not in the Pb-Pb MC I'm using for testing (converted on Aug. 5th). Then, if possible I would like to keep the track selection as it is in order to play a bit with the cuts in a easier way. My objective is to compare TPC tracking efficiency in the new and old (Run2) framework.

Reply:
It's also very nice to use in the near future the trackselection workflow but I didn't find instructions for it.

Reply:
You can join the hyperloop channel for the analysis challenge and ask there. The track selection workflow will add the table you were missing. In a nutshell you just add o2-analysis-trackselection to your workflow.

Reply:
ok, thanks! I'm already in the channel, I will ask for more details there. In the meantime I would like to use this track selection if possible as anticipated in the previous message.

Reply:
I don't agree with this approach. The code is more or less copy-pasted from the TrackSelection class. It is better to use the TrackSelection class (no need to use the default cuts), and configure them as you like. You should be able to do any cut you like. @mkruegerGitHub may comment as well.

Reply:
ok, this will require some time since I would like to test few things. I'll be back to you asap.

Reply:
I also believe that we should avoid duplicating code that already exists esewhere. Documenation for the track selection is already in preparation.
In addition, the code for calculating the dca that was copied here is not entirely correct (see https://github.com/AliceO2Group/AliceO2/pull/4450).

Reply:
thanks, the documentation is really appreciated.

Reply:
Hi Ivan,
the documentation is now available. You can find it here: https://aliceo2group.github.io/analysis-framework/tracksel.html.
Cheers,
  Mario